packages/flutter_tools/lib/src/android/android_emulator.dart

```diff
@@ -5,36 +5,131 @@
 import 'dart:async';
 
 import 'package:meta/meta.dart';
+import 'package:process/process.dart';
 
 import '../android/android_sdk.dart';
 import '../android/android_workflow.dart';
 import '../base/common.dart';
 import '../base/file_system.dart';
 import '../base/io.dart';
+import '../base/logger.dart';
 import '../base/process.dart';
 import '../base/utils.dart';
 import '../convert.dart';
 import '../device.dart';
 import '../emulator.dart';
-import '../globals.dart' as globals;
 import 'android_sdk.dart';
 
 class AndroidEmulators extends EmulatorDiscovery {
+  AndroidEmulators({
+    @required AndroidSdk androidSdk,
+    @required AndroidWorkflow androidWorkflow,
+    @required FileSystem fileSystem,
+    @required Logger logger,
+    @required ProcessManager processManager,
+  }) : _androidSdk = androidSdk,
+       _androidWorkflow = androidWorkflow,
+       _fileSystem = fileSystem,
+       _logger = logger,
+       _processManager = processManager,
+       _processUtils = ProcessUtils(logger: logger, processManager: processManager);
+
+  final AndroidWorkflow _androidWorkflow;
+  final AndroidSdk _androidSdk;
+  final FileSystem _fileSystem;
+  final Logger _logger;
+  final ProcessManager _processManager;
+  final ProcessUtils _processUtils;
+
   @override
   bool get supportsPlatform => true;
 
   @override
-  bool get canListAnything => androidWorkflow.canListEmulators;
+  bool get canListAnything => _androidWorkflow.canListEmulators;
+
+  @override
+  bool get canLaunchAnything => _androidWorkflow.canListEmulators
```

Comment:
the `emulators` command for listing emulator images is not the same as the `avdmanager` command for launching/creating them.